lib/connection/connection_config.js

```diff
@@ -193,11 +196,41 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo)
     Errors.checkArgumentValid(Util.isNumber(proxyPort),
       ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_PORT);
 
-    proxy =
+    if (Util.exists(proxyUser) || Util.exists(proxyPassword))
+    {
+      // check for missing proxyUser
+      Errors.checkArgumentExists(Util.exists(proxyUser),
+        ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_USER);
+
+      // check for invalid proxyUser
+      Errors.checkArgumentValid(Util.isString(proxyUser),
+        ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_USER);
+
+      // check for missing proxyPassword
+      Errors.checkArgumentExists(Util.exists(proxyPassword),
+        ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_PASS);
+
+      // check for invalid proxyPassword
+      Errors.checkArgumentValid(Util.isString(proxyPassword),
+        ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_PASS);
+
+      proxy =
+      {
+        host: proxyHost,
+        port: proxyPort,
+        user: proxyUser,
+        password: proxyPassword,
+        protocol: proxyProtocol
```

Comment:
Is there any default value for `proxyProtocol`? What happens if the customer does not configure the `proxyProtocol`? Also It would be great if we can add a warning log statement if the customer is using an authenticated proxy but has not set `proxyProtocol` to `https`

Reply:
_Is there any default value for proxyProtocol? What happens if the customer does not configure the proxyProtocol?_
- If no value is set for `proxyProtocol`, then the connector will use HTTP to connect to the proxy server. 

_Also It would be great if we can add a warning log statement if the customer is using an authenticated proxy but has not set proxyProtocol to https_
 - Sounds good, I'll add a warning log for that

Reply:
@sfc-gh-hchaturvedi The warning message is added, please review again. Thanks